test/framework/modules.py

```diff
@@ -414,9 +414,11 @@ def test_modpath_extensions_for(self):
             'module use "%s/Compiler/GCC/4.7.2"' % mod_dir,
             # using prepend-path & quoted
             ' prepend-path MODULEPATH "%s/MPI/GCC/4.7.2/OpenMPI/1.6.4"' % mod_dir,
-            # conditional 'use' on subdirectory in $HOME, e.g. when --subdir-user-modules is used
-            "if { [ file isdirectory %s/modules/Compiler/GCC/4.7.2 ] } {" % os.environ['HOME'],
-            "    module use %s/modules/Compiler/GCC/4.7.2" % os.environ['HOME'],
+            # conditional 'use' on subdirectory in e.g. $HOME, e.g. when --subdir-user-modules is used
+            # we cannot use mod_dir = os.environ['HOME'] here since that may not exist in the test
+            # environment
+            "if { [ file isdirectory %s/Compiler/GCC/4.7.2 ] } {" % mod_dir,
+            "    module use %s/Compiler/GCC/4.7.2" % mod_dir,
```

Comment:
we can redefine `$HOME` in the test to `os.path.join(self.test_prefix, 'HOME')`?